packages/fxa-settings/src/pages/ResetPassword/index.tsx

```diff
@@ -82,43 +81,56 @@ const ResetPassword = ({
     navigate('confirm_reset_password', { replace: true });
   }, [navigate]);
 
-  // Set tooltip error message
-  // const setErrorMessage = (errorText: string) => {
-  //   setEmailErrorText(errorText);
-  // };
-
-  // --TODO: Complete onSubmit handling in FXA-6123--
-  // Verify if provided email is linked to existing account
-  // If yes, logViewEvent (reset-password.submit),
-  // initiate password reset and send password reset link by email
-  // Navigate to confirm_reset_password
-  // Catch errors:
-  // If 'UNKNOWN_ACCOUNT' --> tooltip ('Unknown account. Sign up' w/link), log error
-  // If USER_CANCELLED_LOGIN --> log error (login.cancelled)
-  // If other error --> alertBar, ''Sorry, there was a problem resetting your password'
-  const onSubmit = () => {
+  const onSubmit = async () => {
     try {
-      // TODO resetPassword function in FXA-6123
-      account.resetPassword(email);
+      setErrorText('');
+      setUnknownAccountError(false);
+      await account.resetPassword(email);
       navigateToConfirmPwReset();
-    } catch (e) {
-      const errorResetPassword = ftlMsgResolver.getMsg(
-        'reset-password-error-general',
-        'Sorry, there was a problem resetting your password'
-      );
-      alertBar.error(errorResetPassword);
+    } catch (err) {
+      if (err.errno === AuthUiErrors.UNKNOWN_ACCOUNT.errno) {
+        setUnknownAccountError(true);
+      } else {
+        const errorText = ftlMsgResolver.getMsg(
+          composeAuthUiErrorTranslationId(err),
+          'Sorry, there was a problem resetting your password'
```

Comment:
Should this type of error also go in the banner? The `errorText` passed to the input will show up in a tooltip. From what I see in the current content-server version, I think the tooltip is only used if the email input was left blank?

Reply:
Good call out, agreed this should render `<Banner>`.